engines/cine/metaengine.cpp

```diff
@@ -245,6 +252,161 @@ SaveStateDescriptor CineMetaEngine::querySaveMetaInfos(const char *target, int s
 	return SaveStateDescriptor();
 }
 
+Common::KeymapArray CineMetaEngine::initKeymaps(const char *target) const {
+	using namespace Common;
+	using namespace Cine;
+
+	Keymap *engineKeyMap = new Keymap(Keymap::kKeymapTypeGame, "cine-main", "CINE main");
+	Keymap *gameKeyMap = new Keymap(Keymap::kKeymapTypeGame, "game-shortcuts", _("Game Keymappings"));
+	Keymap *mouseKeyMap = new Keymap(Keymap::kKeymapTypeGame, "mouse-shortcuts", _("Key to Mouse Keymappings"));
+
+	Action *act;
+
+	act = new Action(kStandardActionLeftClick, _("Left Click"));
+	act->setLeftClickEvent();
+	act->addDefaultInputMapping("MOUSE_LEFT");
+	act->addDefaultInputMapping("JOY_A");
+	engineKeyMap->addAction(act);
+
+	act = new Action(kStandardActionRightClick, _("Right Click"));
+	act->setRightClickEvent();
+	act->addDefaultInputMapping("MOUSE_RIGHT");
+	act->addDefaultInputMapping("JOY_B");
+	engineKeyMap->addAction(act);
+
+	act = new Action("MOUSELEFT", _("Mouse Left Keys"));
+	act->setCustomEngineActionEvent(kActionMouseLeft);
+	act->addDefaultInputMapping("RETURN");
+	act->addDefaultInputMapping("KP_ENTER");
+	act->addDefaultInputMapping("KP5");
+	mouseKeyMap->addAction(act);
+
+	act = new Action("MOUSERIGHT", _("Mouse Right Keys"));
+	act->setCustomEngineActionEvent(kActionMouseRight);
+	act->addDefaultInputMapping("ESCAPE");
+	mouseKeyMap->addAction(act);
+
+	act = new Action("DEFAULTSPEED", _("Default Speed"));
+	act->setCustomEngineActionEvent(kActionGameSpeedDefault);
+	act->addDefaultInputMapping("KP0");
+	gameKeyMap->addAction(act);
+
+	act = new Action("SLOWERSPEED", _("Slower Speed"));
+	act->setCustomEngineActionEvent(kActionGameSpeedSlower);
+	act->addDefaultInputMapping("MINUS");
+	act->addDefaultInputMapping("KP_MINUS");
+	act->allowKbdRepeats();
+	gameKeyMap->addAction(act);
+
+	act = new Action("FASTERSPEED", _("Faster Speed"));
+	act->setCustomEngineActionEvent(kActionGameSpeedFaster);
+	act->addDefaultInputMapping("PLUS");
+	act->addDefaultInputMapping("KP_PLUS");
+	act->addDefaultInputMapping("C+EQUALS");
```

Comment:
Is this mapping new? I am not seeing a control-equals handling in existing code.

Reply:
Sorry, it was meant to be Shift-Equals, Fixed